pkg/logs/client/http/destination.go

```diff
@@ -102,50 +121,122 @@ func errorToTag(err error) string {
 	}
 }
 
-// Send sends a payload over HTTP,
-// the error returned can be retryable and it is the responsibility of the callee to retry.
-func (d *Destination) Send(payload []byte) error {
-	if d.blockedUntil.After(time.Now()) {
-		log.Debugf("%s: sleeping until %v before retrying", d.url, d.blockedUntil)
-		d.waitForBackoff()
-	}
+// Start starts reading the input channel
+func (d *Destination) Start(input chan *message.Payload, output chan *message.Payload) (stopChan chan struct{}) {
+	stopChan = make(chan struct{})
+	go d.run(input, output, stopChan)
+	return stopChan
+}
 
-	err := d.unconditionalSend(payload)
+func (d *Destination) run(input chan *message.Payload, output chan *message.Payload, stopChan chan struct{}) {
+	var startIdle = time.Now()
 
-	if _, ok := err.(*client.RetryableError); ok {
-		d.nbErrors = d.backoff.IncError(d.nbErrors)
-	} else {
-		d.nbErrors = d.backoff.DecError(d.nbErrors)
+	for p := range input {
+		d.expVars.AddFloat(expVarIdleMsMapKey, float64(time.Since(startIdle)/time.Millisecond))
+		var startInUse = time.Now()
+
+		d.sendConcurrent(p, output)
+
+		d.expVars.AddFloat(expVarInUseMapKey, float64(time.Since(startInUse)/time.Millisecond))
+		startIdle = time.Now()
 	}
+	// Wait for any pending concurrent sends to finish or terminate
+	d.wg.Wait()
 
-	d.blockedUntil = time.Now().Add(d.backoff.GetBackoffDuration(d.nbErrors))
+	d.Lock()
+	d.isRetrying = false
+	d.Unlock()
+	stopChan <- struct{}{}
```

Comment:
I wonder if we need to manage what happens when we have a shutdown sequence and this line sends to a stopChan and then the Destination object owner does the same.

Reply:
that shouldn't really happen unless someone intentionally misuses this. Though I'm open to suggestions to make this safer (and ideally not more complex). 

The simplest alternative I can think of is to give it a stop method that blocks and internalizes the channel logic - this was just simpler given the usage context.

Reply:
I made it directional